.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "e2e/playwright/mattermost-plugin-e2e-test-utils"]
+	path = e2e/playwright/mattermost-plugin-e2e-test-utils
+	url = https://github.com/mattermosttest/mattermost-plugin-e2e-test-utils.git
```

Comment:
I don't love seeing a submodule getting used here. It adds a lot of complexity. Are there simpler approaches that you considered?

Reply:
@hanzei Yes the other two options I've considered is referencing through npm, and just duplicating the code. I definitely don't want to go down the route of duplicating the code. Publishing to npm or referencing as a git-based node dependency is probably what we will end up doing.

The submodule approach was the quickest way that required no hassling with npm. Even if I didn't end up running into an issue with npm, I didn't want that to get in the way when trying this out. A git-based npm dependency wouldn't be much different as far as "use this commit", but I agree the submodule is not as turnkey as referencing a git-based dependency